src/eventloop.jl

```diff
@@ -52,7 +60,72 @@ function waitloop()
     end
 end
 
-const idle_delay = Ref(0.5)
+const idle_delay = Ref(0)
 function set_idle_delay(delay_secs::Float64=0.5)
     idle_delay[] = delay_secs
 end
+
+#=
+    Hold Channel Interface
+    This allows user processes to register and deregister
+    channels that allow user code to hold IJulia from sending
+    the idle status message until signaled.  This would have
+    been prettier if there were level-triggered events in
+    Julia base, but this will do okay for now.
+
+    Users should reset their own channels as IJulia will not
+    reach into their channel state manually unless there is a
+    timeout, in which case a temporary variable is put onto the channel
+    buffer and removed after the wait process.
+    See wait_multi_with_timeout()
+=#
+const hold_channels = Array{Channel{Bool},1}()
+const hold_channels_timeout_default = 3
+function add_hold_channel(channel::Channel{Bool})
+    if !(channel in hold_channels)
+        hold_channels = [hold_channels..., channel]
+    end
+end
+
+function remove_hold_channel(channel::Channel{Bool})
+    if channel in hold_channels
+        filter!(e->e∉[channel],hold_channels)
+    end
+end
+
+function wait_multi_with_timeout(channels::Array{Channel{Bool},1}, timeout::Number)
+    notification = Channel{Bool}(1)
+    #=
+        This async routine will manually fulfill any channels (aka semaphores)
+        that don't come through before the timeout.  Doing this because someone
+        will inevitably not fulfill the semaphores that they register, and
+        thus this will take care of us, at least for the short term.
+    =#
+    @async begin
+        sleep(timeout)
+        idxs_to_reset = []
+        for i in eachindex(channels)
+            if length(channels[i].data) == 0
+                put!(false)
+                idxs_to_reset = [idxs_to_reset..., i]
+            end
+        end
+        wait(notification)
+        for i in idxs_to_reset
+            take!(channels[i])
+        end
+    end
+    for channel in channels
+        wait(channel)
+    end
+    put!(notification, true)
+end
+
+#= Interact.jl compatibility code
+    This ensures the existing fix continues to work with Interact.jl until
+    they can update to using the new semaphore or channel interface.
+=#
+const interact_min_delay = 0.25
+function check_interact_loaded()
+    return isdefined(:Interact)
```

Comment:
I'm okay with that.  It is a little more general to say isdefined(:Interact), but if you want to restrict it to Main that's okay too.  The penalty for a false positive here isn't huge, at least in the sense that it's unlikely someone would define Interact and care about response performance.